gnovm/stdlibs/std/coins.gno

```diff
@@ -13,28 +14,147 @@ type Coin struct {
 	Amount int64  `json:"amount"`
 }
 
+// NewCoin returns a new coin with a denomination and amount
+func NewCoin(denom string, amount int64) Coin {
+	return Coin{
+		Denom:  denom,
+		Amount: amount,
+	}
+}
+
+// String provides a human-readable representation of a coin
 func (c Coin) String() string {
+	if c.IsZero() {
+		return "0" + c.Denom
+	}
+
 	return strconv.Itoa(int(c.Amount)) + c.Denom
 }
 
+// IsGTE returns true if they are the same type and the receiver is
+// an equal or greater value
 func (c Coin) IsGTE(other Coin) bool {
 	if c.Denom != other.Denom {
-		panic("invalid coin denominations: " + c.Denom)
+		panic("comparing different denominations: " + c.Denom + ", " + other.Denom)
 	}
+
 	return c.Amount >= other.Amount
 }
 
+// IsLT returns true if they are the same type and the receiver is
+// a smaller value
+func (c Coin) IsLT(other Coin) bool {
+	if c.Denom != other.Denom {
+		panic("comparing different denominations: " + c.Denom + ", " + other.Denom)
+	}
+
+	return c.Amount < other.Amount
+}
+
+// IsEqual returns true if the two sets of Coins have the same value
+func (c Coin) IsEqual(other Coin) bool {
+	if c.Denom != other.Denom {
+		panic("comparing different denominations: " + c.Denom + ", " + other.Denom)
+	}
```

Comment:
This check should be extracted as it's very common

Reply:
[de5f3a3](https://github.com/gnolang/gno/pull/2104/commits/de5f3a39463310df956c79dbb948cbb32400b600)